zeppelin-web/src/app/visualization/builtins/visualization-d3network.js

```diff
@@ -166,20 +166,20 @@ export default class NetworkVisualization extends Visualization {
 
     const drag = d3.behavior.drag()
       .origin((d) => d)
-      .on('dragstart', function(d) {
+      .on('dragstart', (d) => {
         console.log('dragstart');
         d3.event.sourceEvent.stopPropagation();
         d3.select(this).classed('dragging', true);
         self.force.stop();
       })
-      .on('drag', function(d) {
+      .on('drag', (d) => {
         console.log('drag');
         d.px += d3.event.dx;
         d.py += d3.event.dy;
         d.x += d3.event.dx;
         d.y += d3.event.dy;
       })
-      .on('dragend', function(d) {
+      .on('dragend', (d) => {
         console.log('dragend');
         d.fixed = true;
         d3.select(this).classed('dragging', false);
```

Comment:
Doesn’t `this` work differently between these functions? In event handlers, `this` is expected to refer to the DOM element. As far as I know, arrow functions do not bind `this` to the DOM element.

Reply:
I think if we need to change it to an arrow function, we should use `d3.event.sourceEvent.target` instead of `this`.
I haven't tested it personally, but I referred to [the D3 Library V3 documentation](https://github.com/d3/d3-3.x-api-reference/blob/master/Selections.md#d3_event).

Reply:
Wow! It's quite new for me. I don't get it completely but it looks like a similar kind of problem with closures and scoping between arrow and function.